Add request body arguemnts to tech docs

RobertM905 opened the pull request:
### Context

The "Request body" part of the template doesn't show what to send in the request body, only an example.

### Changes proposed in this pull request

Add an arguments temple and renderer function to display request_body argument details.

### Guidance to review

![image](https://user-images.githubusercontent.com/47318392/68022985-d7acc580-fc9d-11e9-8b35-9432610aa064.png)

Check argument rendering function and temple to check that the correct information is displayed.
### Release notes

- [ ] [If needed](/README.md#release-notes), the [release notes](/source/release-notes.html.md) have been updated

### Link to Trello card

[1118 - Add request body schema to endpoints](https://trello.com/c/uyXUmycj/1118-add-request-body-schema-to-endpoints)

duncanjbrown commented:
Nice! A couple of design qs.

- `Arguments` feels like it's almost the right thing — the role of this information is to be arguments to a procedure call — but the thing we're describing is a JSON document. Maybe we should be more generic: `Request body schema`?
- We already have schemas for `Rejection` and `Offer`. I think the inclusion of a timestamp on the rejection is a mistake, so I believe we could just reuse these schemas for the requests. I tried changing it in openapi-spec.yml and got this, which is not too bad:

```yml
      requestBody:
        description: The reason for rejection
        content:
          application/json:
            schema:
              type: object
              properties:
                data:
                  $ref: '#/components/schemas/Rejection'
                meta:
                  $ref: '#/components/schemas/TimestampAndAttribution'
```

![Screenshot 2019-11-01 at 16 29 50](https://user-images.githubusercontent.com/642279/68039822-de026800-fcc4-11e9-9f03-325c2353cb1d.png)

- We already define the `meta` as a `TimestampAndAttribution` in the `schema` section, so we could also link to that, but I think this would require some reworking of the code to make the output look like this, ie documenting only the top-level keys

### Request Body Schema

Name   | Type   | Required | Description                 | Schema                  
-------|--------|----------|-----------------------------|-------------------------
data   | object | true     |  | Rejection 
meta   | object | true     |  | TimestampAndAttribution 

What do you think?

RobertM905 commented:
Renamed in https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training-tech-docs/pull/107/commits/a493785225a96b4e3d37c4b4738ac0e7a3ea8b30

And added the reject_application schema change in https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training-tech-docs/pull/108

duncanjbrown commented:
:+1: Thanks. Shall we add something to the changelog?
